### PR TITLE
feat: TODO list

### PR DESCRIPTION
**Pull request by oleksandr-khoma**

### CODE REVIEW

File: homework-todo.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+interface ITodoList {
+    notes: INote[];
+    add(note: INote): void;
+    remove(note: INote): void;
+    get(id: number): INote | undefined;
+    getStats(): { total: number; completed: number; remaining: number };
+    search(query: string): INote[];
+    sort(by: 'status' | 'createdAt'): INote[];
```
Comment (KMaxLSVN) [inline homework-todo.ts]: I don't like this type of function parameter annotation. You should consider improving the type guard for this scenario.

File: homework-todo.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        }
+        this.title = title;
+        this.content = content;
+        this.createdAt = new Date();
+        this.updatedAt = new Date();
+        this.completed = false;
+        this.requiresConfirmation = requiresConfirmation;
+    }
+
+    edit(title: string, content: string) {
```
Comment (KMaxLSVN) [inline homework-todo.ts]: Don't forget to explicitly add type annotations for function return types.

File: homework-todo.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    notes: INote[];
+    add(note: INote): void;
+    remove(note: INote): void;
+    get(id: number): INote | undefined;
+    getStats(): { total: number; completed: number; remaining: number };
+    search(query: string): INote[];
+    sort(by: 'status' | 'createdAt'): INote[];
+}
+
+class TodoList implements ITodoList {
```
Comment (KMaxLSVN) [inline homework-todo.ts:61]: Can't find a method to achieve all ToDos.

There is a method in the Note class to mark a note as completed, but I can't find that functionality in the TodoList class. TodoList is your final application.

File: homework-todo.ts [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    notes: Note[] = [];
+
+    add(note: Note) {
+        this.notes.push(note);
```
Comment (KMaxLSVN) [inline homework-todo.ts:66]: You added validation for an empty note in the Note class constructor. However, you didn't create an instance of Note; you just pushed it. The requirements are different.

File: homework-todo.ts [hunk 5]
```diff
@@ -0,0 +1,97 @@
+interface INote {
```
Comment (KMaxLSVN) [inline homework-todo.ts:3]: There should be two types of notes - with the "requiresConfirmation" field and without it. At least it should be an optional field. Please, do not attempt to implement the simpler solution.

File: homework-todo.ts [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    getStats() {
+        const total = this.notes.length;
+        const completed = this.notes.filter(note => note.completed).length;
+        const remaining = total - completed;
+        return { total, completed, remaining };
+    }
+
+    search(query: string) {
```
Comment (KMaxLSVN) [inline homework-todo.ts]: The functionality should be implemented separately by extending. This is indeed what the requirements call for.

File: homework-todo.ts [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        return this.notes.filter(note => note.title.includes(query) || note.content.includes(query));
+    }
+
+    sort(by: 'status' | 'createdAt') {
```
Comment (KMaxLSVN) [inline homework-todo.ts]: The functionality should be implemented separately by extending. This is indeed what the requirements call for.

File: homework-todo.ts (hunk 3 above)
Comment (KMaxLSVN) [reply homework-todo.ts:61]: Still unable to find a method for fetching all Todos. Add all necessary fields and methods to the ITodoList interface to ensure comprehensive implementation.

Comment (oleksandr-khoma) [reply homework-todo.ts:61]: @KMaxLSVN added, thnx

**Pull request merged**